editor/modes/visual-editor/block.js

```diff
@@ -11,84 +11,116 @@ import Toolbar from 'components/toolbar';
 import BlockMover from 'components/block-mover';
 import BlockSwitcher from 'components/block-switcher';
 
-function VisualEditorBlock( props ) {
-	const { block } = props;
-	const settings = wp.blocks.getBlockSettings( block.blockType );
+class VisualEditorBlock extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.bindBlockNode = this.bindBlockNode.bind( this );
+	}
 
-	let BlockEdit;
-	if ( settings ) {
-		BlockEdit = settings.edit || settings.save;
+	bindBlockNode( node ) {
+		this.node = node;
 	}
 
-	if ( ! BlockEdit ) {
-		return null;
+	componentWillReceiveProps( newProps ) {
+		if (
+			this.props.order !== newProps.order &&
+			this.props.isSelected &&
+			newProps.isSelected
+		) {
+			this.previousOffset = this.node.getBoundingClientRect().top;
```

Comment:
No, actually we don't want to keep the same `scrollY` since the block is moving, we want the block to show in the same position.